pandas/formats/style.py

```diff
@@ -851,39 +851,178 @@ def set_properties(self, subset=None, **kwargs):
+        ----------
+        color: 2-tuple/list, of [``color_negative``, ``color_positive``]
+        width: float
+            A number between 0 or 100. The largest value will cover ``width``
+            percent of the cell's width
+        base: str
+            The base css format of the cell, e.g.:
+            ``base = 'width: 10em; height: 80%;'``
+
+        Returns
+        -------
+        self : Styler
+        """
+
+        if s.min() >= 0:
+            # In this case, we place the zero at the left, and the max() should
+            # be at width
+            zero = 0.0
+            slope = width / s.max()
+        elif s.max() <= 0:
+            # In this case, we place the zero at the right, and the min()
+            # should be at 100-width
+            zero = 100.0
+            slope = width / -s.min()
+        else:
+            slope = width / (s.max() - s.min())
+            zero = (100.0 + width) / 2.0 - slope * s.max()
+
+        normed = zero + slope * s
+
+        attrs_neg = (base + 'background: linear-gradient(90deg, transparent 0%'
+                     ', transparent {w:.1f}%, {c} {w:.1f}%, '
+                     '{c} {zero:.1f}%, transparent {zero:.1f}%)')
+
+        attrs_pos = (base + 'background: linear-gradient(90deg, transparent 0%'
+                     ', transparent {zero:.1f}%, {c} {zero:.1f}%, '
+                     '{c} {w:.1f}%, transparent {w:.1f}%)')
+
+        return [attrs_pos.format(c=color[1], zero=zero, w=x) if x > zero
+                else attrs_neg.format(c=color[0], zero=zero, w=x)
+                for x in normed]
+
+    def bar(self, subset=None, align='left', axis=0,
+            color='#d65f5f', width=100):
         """
         Color the background ``color`` proptional to the values in each column.
         Excludes non-numeric data by default.
-
         .. versionadded:: 0.17.1
 
         Parameters
         ----------
         subset: IndexSlice, default None
             a valid slice for ``data`` to limit the style application to
         axis: int
-        color: str
+        color: str (for align='left') or 2-tuple/list (for align='zero', 'mid')
+            If a str is passed, the color is the same for both
+            negative and positive numbers. If 2-tuple/list is used, the
+            first element is the color_negative and the second is the
+            color_positive (eg: ['#d65f5f', '#5fba7d'])
         width: float
             A number between 0 or 100. The largest value will cover ``width``
             percent of the cell's width
+        align : str, default 'left'
+            
+            .. versionadded:: 0.20.0
+            
+            - 'left' : the min value starts at the left of the cell
+            - 'zero' : a value of zero is located at the center of the cell
+            - 'mid' : the center of the cell is at (max-min)/2, or
+                if values are all negative (positive) the zero is aligned
+                at the right (left) of the cell
```

Comment:
@jorisvandenbossche: @TomAugspurger is correct, it's really centered on zero. The example I tried to include in the doc and that I pasted in the first post of this PR should highlight that fact, otherwise I should revise it.

I have no idea how you handle deprecation cycles etc, so I'll leave that completely up to you guys.